go-controller/pkg/ovn/gateway_init.go

```diff
@@ -275,7 +275,7 @@ func (oc *Controller) gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet,
 			{
 				Model: &logicalRouterStaticRoute,
 				ModelPredicate: func(lrsr *nbdb.LogicalRouterStaticRoute) bool {
-					return lrsr.OutputPort != nil && *lrsr.OutputPort == externalRouterPort && lrsr.Nexthop == nextHop.String()
+					return lrsr.OutputPort != nil && *lrsr.OutputPort == externalRouterPort && lrsr.IPPrefix == allIPs
```

Comment:
really this should also not match on output port. If the output port changed somehow (I don't think we support this today) to the secondary bridge then we would need to update the output port as well. Really prefix is the only thing we should care about. Can fix this later though.